### ISSUE TITLE
Help! I can't install pyshtools!!!

### ISSUE THREAD

**Issue by MarkWieczorek** [SHTOOLS/SHTOOLS#459]
**Description of the problem**

**System information**

**Comment by MarkWieczorek** [comment]
I have started the migration from multibuild to cibuildwheel, which should hopefully solve some of our problems with distributing wheels on pypi: https://github.com/SHTOOLS/SHTOOLS/pull/469

It looks like this is working for linux and macos (both intel and apple silicon). These will be uploaded with the next release in about a month or so. However, I have no idea on how to build wheels for windows, so if you know anything about this, your help would be appreciated! I also would really like to make aarch64 builds, but this involves cross compiling, which complicates installing fftw and openblas.

**Comment by cometbeetle** [comment]
If you're still using AppVeyor to build the Windows wheels, I think I found a solution. It involves using the `MSYS2` that comes with the AppVeyor image, and then installing the dependencies (`Fortran`, `fftw3`, `openblac`, etc.) using `pacman`. It then adds the DLLs to the built wheel using [delvewheel](https://github.com/adang1345/delvewheel). This takes a bit longer, but seems to work reliably and makes the `appveyor.yml` configuration a bit cleaner. Note that I disabled the uploads to PyPi, however they would work with the proper credentials. 

Here is the [link to my fork](https://github.com/cometbeetle/build-shtools/tree/testing-windows) with the updated appveyor.yml file. I'm pretty new to this, so Things like caching could be improved with my configuration. I also imagine a similar approach (i.e., using MSYS2) could work with GitHub actions.

**Comment by cometbeetle** [comment]
Also, would there be any way to upload the MacOS arm64 wheel to PyPi (I believe it's already on Anaconda)?

**Comment by MarkWieczorek** [comment]
Thanks for looking into this!

First, could you take a look at this PR? https://github.com/SHTOOLS/SHTOOLS/pull/469 I'm trying to get away from our current multibuild/appveyor wheel builders, as these only cause me continuous grief. Linux and macOS (ARM+intel) are working, but windows isn't. I would like to get linux ARM to work too, but this is of less importance.

As for the MacOS arm64 builds, we upload everything directly to pypi, and as far as I know, there isn't an arm64 build; this is part of the reason for trying to transition to cibuildwheel. If there is an ARM64 build, I doubt it works.

Lastly, I would like to put out a new release before the end of July, but I was hoping to include the open PR regarding Boule, which is waiting on a new upstream release. I'll ping the devs there, and if this is going to take a long time, I'll just go ahead without them.

**Comment by cometbeetle** [comment]
I've almost got Windows builds working with `cibuildwheel` in #479, however there's a slight irregularity with oldest-supported-`numpy`. I'll continue testing to see if I can get that fixed.

**Comment by cometbeetle** [comment]
Another thing I noticed is that `cibuildwheel` is building for both CPython and PyPy. The `numpy` dependency issue only occurs in the PyPy variant (version 3.10), and I am wondering if you intend to support PyPy Python in addition to plain old CPython. 

For simplicity, I'm going to add the PyPy wheels to CIBW_SKIP. Please let me know if the plan is to support PyPy, though, so I can revert the change. 

**Otherwise, the builds for Windows are good to go!** The only changes from #469 that I made in #479 are to re-enable Windows support. I have also tested importing the wheel it generates, and no DLL errors occur.

**Comment by MarkWieczorek** [comment]
I don't have problems with skipping PyPy. As far as I know, we never built PyPy wheels in the past.

**Comment by MarkWieczorek** [comment]
I am going to close this as we have now (1) successfully transitioned to meson for our build backend, (2) we have successfully migrated to cibuildwheel for building all wheels, and (3) we now are compatible with using numpy 2.0.

If anyone has any further issues, please open a new issue.
